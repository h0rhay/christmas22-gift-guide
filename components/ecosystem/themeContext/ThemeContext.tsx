import React, { ReactChild, ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import theme from "styles/theme";

const ColoredSurface = styled.div`
  background-color: ${({ theme }) => theme.currentColorGroup.surface};
  * {
    color: ${({ theme }) => theme.currentColorGroup.interface};
  }
`;

type ThemeContextProps = {
  colorGroup?: string;
  children: ReactChild | Array<ReactChild> | ReactNode;
};

const ThemeContext = ({
  children,
  colorGroup = theme.fallbackColorGroup,
}: ThemeContextProps) => {
  const currentColorGroup =
    theme.colorsGroup[colorGroup] ||
    theme.colorsGroup[theme.fallbackColorGroup];
  return (
    <ThemeProvider theme={{ ...theme, currentColorGroup }}>
      <ColoredSurface className="colored-surface">{children}</ColoredSurface>
    </ThemeProvider>
  );
};

export default ThemeContext;
