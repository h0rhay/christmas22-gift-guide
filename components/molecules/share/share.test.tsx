import { render, screen, fireEvent, act } from '@testing-library/react';
import Share from '.';
import { ThemeProvider } from 'styled-components';
import Theme from 'styles/theme';
import ThemeContext from "components/ecosystem/themeContext/ThemeContext";

let container: HTMLElement;

const props = {
  type: 'circleButton',
  url: 'foobar.com'
}
const renderComponent = () =>
  render(
    <ThemeProvider theme={Theme}>
      <ThemeContext colorGroup="hero">
        <Share {...props} />
      </ThemeContext>
    </ThemeProvider>
  );

describe('Share', () => {
  beforeEach(() => {
    container = renderComponent().container;
  });

  it('should match screenshot', () => {
    expect(container).toMatchSnapshot();
  });

  describe('When share button is clicked', () => {
    beforeEach(() => {
      fireEvent.click(screen.getByRole('button'));
      Object.assign(window.navigator, {
        clipboard: {
          writeText: jest.fn().mockImplementation(() => Promise.resolve()),
        },
      });
      jest.useFakeTimers();
    });
    it('Modal should appear', () => {
      const modal = screen.getByText(/share this page/i);
      expect(modal).toBeInTheDocument();
      expect(modal).toBeInTheDocument();
    });

    it('Whatsapp share button should appear', () => {
      const whatsapp = screen.getByText(/whatsapp/i);
      expect(whatsapp).toBeInTheDocument();
    });

    it('Facebook share button should appear', () => {
      const facebook = screen.getByText(/facebook/i);
      expect(facebook).toBeInTheDocument();
    });

    it('Twitter share button should appear', () => {
      const twitter = screen.getByText(/twitter/i);
      expect(twitter).toBeInTheDocument();
    });

    it('Linkedin share button should appear', () => {
      const linkedin = screen.getByText(/linkedin/i);
      expect(linkedin).toBeInTheDocument();
    });

    it('Copy Link button should appear', () => {
      expect(screen.getByText(/copy link/i)).toBeInTheDocument();

      act(() => {
        const copyLinkButton = screen.getByTestId('copy-link-button');
        fireEvent.click(copyLinkButton);
      });

      expect(screen.getByText(/link copied/i)).toBeInTheDocument();
      expect(screen.queryByText(/copy link/i)).not.toBeInTheDocument();

      act(() => jest.advanceTimersByTime(5000));

      expect(screen.getByText(/copy link/i)).toBeInTheDocument();
      expect(screen.queryByText(/link copied/i)).not.toBeInTheDocument();
    });
  });
});