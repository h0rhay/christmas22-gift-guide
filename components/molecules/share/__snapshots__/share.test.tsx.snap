// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share should match screenshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <button
      class="circleButton__Wrapper-sc-o294s7-0 gxqiFO"
    >
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
        />
        <polyline
          points="16 6 12 2 8 6"
        />
        <line
          x1="12"
          x2="12"
          y1="2"
          y2="15"
        />
      </svg>
    </button>
  </div>
</div>
`;
